aea/contracts/ethereum.py

```diff
@@ -73,17 +73,18 @@ def set_instance(self, ledger_api: EthereumApi) -> None:
         :param ledger_api: the ethereum ledger api
         """
         assert self._instance is None, "Instance already set!"
-        self._instance = ledger_api.api.web3.Contract(
+        self._instance = ledger_api.api.eth.contract(
             abi=self.abi, bytecode=self.bytecode
         )
 
-    def set_address(self, ledger_api: EthereumApi, contract_address: str) -> None:
+    def set_address(self, ledger_api: EthereumApi, tx_digest: str) -> None:
```

Comment:
For separation of concerns I'd keep this as previously implemented.

Reply:
But I will not be able to get the contract address...